tests/event.rs

```diff
@@ -0,0 +1,34 @@
+use calendar_backend::run;
+
+#[tokio::test]
+async fn create_event_works() {
+
+    // Arrange
+    spawn_app();
+
+    let client = reqwest::Client::new();
+    let event = serde_json::json!({
+        "name": "write_integration_tests",
+        "calendar_id": "djacota",
+        "date_time": "09-05-2023"
+    });
+
+    // Act
+    let response = client
+        .post("http://127.0.0.1:8080/event")
+        .json(&event)
+        .send()
+        .await
+        .expect("Failed to execute request.");
+
+    // Assert
+    assert!(response.status().is_success());
+    assert_eq!(Some(0), response.content_length());
+}
+
+// launch the server as a background task
+fn spawn_app() {
+    let server = run().expect("Failed to bind address");
+    let _ = tokio::spawn(server);
```

Comment:
Is this running instance of `calendar-backend` killed at the end of the test?

Reply:
In [this documentation](https://docs.rs/tokio/latest/tokio/task/fn.spawn.html) is specified that:

> When a runtime is shutdown, all outstanding tasks are dropped, regardless of the lifecycle of that task.

Not the most accurate source, but ChatGPT says:

> Here's how multiple tests with #[tokio::test] work:
> 
> 1. **Sequential Execution**: The test functions annotated with #[tokio::test] within the same module will be executed one at a time, in the order they are defined in the code.
> 
> 2. **Shared Tokio Runtime**: All of these tests will share the same instance of the Tokio asynchronous runtime. The Tokio runtime is set up once before the first test runs and is torn down once after the last test completes.
> 
> 3. **Isolation**: Each test is isolated from the others in terms of state and resource management. Any changes made to the state or resources within one test should not affect the others.
> 
> 4. **Cleanup**: The Tokio runtime is responsible for cleaning up any asynchronous tasks, resources, and other runtime-specific components after all tests have completed.